oneview/data_source_server_profile_template.go

```diff
@@ -1070,7 +1072,9 @@ func dataSourceServerProfileTemplateRead(d *schema.ResourceData, meta interface{
 		"manage_firmware":          spt.Firmware.ManageFirmware,
 	})
 	d.Set("firmware", firmwareOption)
-	if len(spt.OSDeploymentSettings.OSCustomAttributes) != 0 {
+
+	OsDeploymentSetting := ov.OSDeploymentSettings{}
+	if reflect.DeepEqual(spt.OSDeploymentSettings, OsDeploymentSetting) == false {
```

Comment:
Is reflect.DeepEqual used for comparing nested objects?

Reply:
yes